core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/UserStoreDeploymentManager.java

```diff
@@ -63,23 +63,23 @@ public void deploy(String absoluteFilePath) throws DeploymentException {
                 //tenant admin modified secondary user store configuration
                 if (filePathSegments[filePathSegments.length - 4].equals(UserStoreConfigConstants.TENANTS)) {
                     realmConfiguration.setTenantId(tenantId);
-                    setSecondaryUserStore(userRealm.getRealmConfiguration(), realmConfiguration);
-                    primaryUSM = (AbstractUserStoreManager) userRealm.getUserStoreManager();
-                    primaryUSM.addSecondaryUserStoreManager(realmConfiguration, userRealm);
+                }
 
-                } else {
-                    //super tenant modified secondary user store configuration
-                    setSecondaryUserStore(userRealm.getRealmConfiguration(), realmConfiguration);
-                    primaryUSM = (AbstractUserStoreManager) userRealm.getUserStoreManager();
+                setSecondaryUserStore(userRealm.getRealmConfiguration(), realmConfiguration);
+                primaryUSM = (AbstractUserStoreManager) userRealm.getUserStoreManager();
+                try {
                     primaryUSM.addSecondaryUserStoreManager(realmConfiguration, userRealm);
+                } catch (UserStoreException ex) {
+                    log.warn(ex.getMessage());
```

Comment:
Do we need to add a warn log here as we are throwing the exception ?

Reply:
removed the warn log in https://github.com/wso2/carbon-kernel/pull/2550